plugins/lsp/conform.nix

```diff
@@ -0,0 +1,90 @@
+{
+  pkgs,
+  config,
+  lib,
+  ...
+}:
+with lib; let
+  cfg = config.plugins.conform;
+  helpers = import ../helpers.nix {inherit lib;};
+in {
+  options.plugins.conform =
+    helpers.extraOptionsOptions
+    // {
+      enable = mkEnableOption "conform.nvim";
+
+      package = helpers.mkPackageOption "conform.nvim" pkgs.vimPlugins.conform-nvim;
+
+      # Map of filetype to formatters
+      formattersByFt =
+        helpers.defaultNullOpts.mkNullable types.attrs
+        "see documentation"
+        ''
+          Map of filetype to formatters
+          lua = { "stylua" },
+          Conform will run multiple formatters sequentially
+          python = { "isort", "black" },
+          Use a sub-list to run only the first available formatter
+          javascript = { { "prettierd", "prettier" } },
+          Use the "*" filetype to run formatters on all filetypes.
+          ["*"] = { "codespell" },
+          Use the "_" filetype to run formatters on filetypes that don't
+          have other formatters configured.
+          ["_"] = { "trim_whitespace" },
+        '';
+      # If this is set, Conform will run the formatter on save.
+      # It will pass the table to conform.format().
+      # This can also be a function that returns the table.
+      # See :help conform.format for details.
+      formatOnSave = {
+        lspFallback =
+          helpers.defaultNullOpts.mkBool true
+          "See :help conform.format for details.";
+        timeoutMs =
+          helpers.defaultNullOpts.mkNullable (types.int) 1000
+          "See :help conform.format for details.";
+      };
+
+      # If this is set, Conform will run the formatter asynchronously after save.
+      # It will pass the table to conform.format().
+      # This can also be a function that returns the table.
+      formatAfterSave = {
+        lspFallback =
+          helpers.defaultNullOpts.mkBool true
+          "See :help conform.format for details.";
+      };
+
+      # Set the log level. Use `:ConformInfo` to see the location of the log file.
+      logLevel =
+        helpers.defaultNullOpts.mkNullable
+        types.str
+        "vim.log.levels.ERROR"
+        " See :h log_levels ";
```

Comment:
You can do the same thing as [here](https://github.com/nix-community/nixvim/blob/0c5dcb56d105addedd001f043e6a32ab9f1abe92/plugins/languages/treesitter/rainbow-delimiters.nix#L213-L218)

Reply:
Is this correct now:
```nix
      logLevel =
        helpers.ifNonNull' level
        (helpers.mkRaw "vim.log.levels.${strings.toUpper level}");
```

Reply:
It seems correct to me

Reply:
> It seems correct to me

@GaetanLepage pointed out on matrix that this isn't how it should be done. This is what they said:

> Hmmm
Not really
here is where you declare your option, so no "vim.log.levels." in front
Have I sent you an example ?

I suppose he meant that the options should not be hardcoded like this and it should only state the type.